--- external/iers/src_2010/libiers-hardisp/makefile.orig	2025-09-10 13:40:27.028757770 +0200
+++ external/iers/src_2010/libiers-hardisp/makefile	2025-09-10 13:40:44.361618083 +0200
@@ -22,13 +22,13 @@
 # here.
 
 FC = gfortran 
-FF = -c -O 
+FF = -c -O -fPIC
 
 #----YOU SHOULDN'T HAVE TO MODIFY ANYTHING BELOW THIS LINE---------
 
 # Name the IERS library in its source location.

-IERS_LIB = libiers-hardisp.a
+IERS_LIB = libiers_hardisp.a

 # The list of IERS library object files.

